### DISCUSSION TITLE
yarn not found in VSCode 1.74.0 and plugin version v5.0.3

### DISCUSSION THREAD

**Discussion by pearj** [orta/vscode-jest#964] [Q&A] [answered]
vscode-jest was happily working until I just updated to VSCode 1.74.0, and now for some reason it can't find jest.

I tried specifying the `jest.jestCommandLine` and `jest.rootPath` in `settings.json` but it didn't make a difference
```json
  "jest.jestCommandLine": "yarn test",
  "jest.rootPath": "/home/pearj/dev/forgeops/typescript/idm"
```

The output from the jest terminal
```
Jest (forgeops)
/bin/sh: 1: yarn: not found

/bin/sh: 1: yarn: not found


[error] failed to retrieve test file list. TestExplorer might show incomplete test items

[info] jest process failed to start, most likely due to env or project configuration issues, please see: https://github.com/jest-community/vscode-jest/blob/master/README.md#jest-failed-to-run

[error] Jest process "watch-tests" ended unexpectedly

[info] jest process failed to start, most likely due to env or project configuration issues, please see: https://github.com/jest-community/vscode-jest/blob/master/README.md#jest-failed-to-run
```

Does anyone have any idea what might have happened?

**Comment by connectdotz** [comment, marked as answer]
Hmm... this is strange... looks like it is an env issue, but it should fall back to the login shell in that case. If you still have the console log, can you see if it is running a login shell? (You can search for `spawning login-shell` in the console log if the debug mode is enabled.)

What happens when you reload the window or restart vscode?

**Reply by pearj** [reply]
So my PC rebooted overnight, and now it's ok. I am running VSCode in WSL2 so maybe something went a bit weird when vscode upgraded, and I just needed to make sure all the old vscode/node processes had stopped or just needed to restart WSL2.

**Comment by GeorgeWL** [comment]
2 years later, I've just started seeing this same error

> [error] Failed to start jest session: 'yarn' is not recognized as an internal or external command,
operable program or batch file.

running vscode in user mode, no WSL, Git Bash for Windows as default terminal
